app/models/ncr/work_order.rb

```diff
@@ -24,21 +26,81 @@ class WorkOrder < ActiveRecord::Base
     validates :building_number, presence: true
     validates :office, presence: true
 
-    # Ignore values in these fields, depending on the expense type
-    def ignore_fields
-      case self.expense_type
+    def set_defaults
+      self.not_to_exceed ||= false
+      self.emergency ||= false
+    end
+
+    # @todo - this is an awkward dance due to the lingering Cart model. Remove
+    # that dependence
+    def create_cart(approver_email, description, requester)
+      cart = Cart.create(
+        name: description,
+        proposal_attributes: {flow: 'linear', client_data: self}
+      )
+      cart.set_requester(requester)
+      self.add_approvals_on(cart, approver_email)
+      Dispatcher.deliver_new_cart_emails(cart)
+      cart
+    end
+    def update_cart(approver_email, description, cart)
+      cart.name = description
+      cart.approver_approvals.destroy_all
+      self.add_approvals_on(cart, approver_email)
+      cart.restart!
+      cart
+    end
+
+    def system_approvers
+      emails = [ENV['NCR_BUDGET_APPROVER_EMAIL'] ||
+                'communicart.budget.approver@gmail.com']
+      if self.expense_type == 'BA61'
+        emails << (ENV['NCR_FINANCE_APPROVER_EMAIL'] ||
+                   'communicart.ofm.approver@gmail.com')
+      else
+        emails
+      end
+    end
+
+    def add_approvals_on(cart, approver_email)
```

Comment:
The WorkOrder already has the association, so shouldn't need the Cart to be passed in.